### ISSUE TITLE
PoincareAtoms decode/exp saturates ONTO the ball boundary (‖x‖==1) for large tangents → distance/log degenerate

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#354]
## Summary
`gamfit.torch.PoincareAtoms.forward` (and the underlying exp map) returns points with norm **exactly 1.0** — on the boundary of the open Poincare ball, not strictly inside — when the aggregated tangent is large. Boundary points are not valid elements of the ball model, so downstream `distance`/`log` operations degenerate (blow up / become ill-defined).

## Repro (gamfit 0.1.134, PyPI wheel)
```python
import torch
from gamfit.torch import PoincareAtoms
pa = PoincareAtoms(F=4, ball_dim=6, curvature=-1.0)
x = pa(torch.full((3, 4), 1e3))      # large gate -> large aggregated tangent
print("norms:", [round(float(v), 8) for v in x.norm(dim=-1)])
print("strictly inside (<1)?", bool((x.norm(dim=-1) < 1.0).all()))
print("dist(0,x):", [float(v) for v in pa.distance(torch.zeros_like(x), x)])
```
Output:
```
norms: [1.0, 1.0, 1.0]
strictly inside (<1)? False
dist(0,x): [35.92, 35.92, 35.92]
```

## Expected
The exp/decode must guarantee a strictly-interior result (`||x|| <= (1 - eps)/sqrt(c)`). For large `||v||` the exp map mathematically approaches but never reaches the boundary; float saturation to exactly 1.0 should be clamped so the output stays a valid open-ball point and `log`/`distance` remain finite and well-defined.

## Impact
Any decode of a confidently-activated atom can land on the boundary, making subsequent geodesic ops degenerate. Worked around downstream by rescaling outputs to `(1 - 1e-6)/sqrt(c)`, but the primitive should enforce the open-ball invariant itself.

**Comment by SauersML** [comment]
Closing — landed fix `2cb934464` meets the quality bar.

**Root cause, principled — enforced at every exp site.** The boundary clamp is baked into the radial exp coefficient `exp_coeff(s) = min(tanh(s), 1-BOUNDARY_EPS)/s` (src/geometry/poincare.rs:80), applied at `exp_origin` and `tangent_decode_forward`. The Lorentz path caps the lift argument at `EXP_SATURATION_CAP` (preventing cosh/sinh overflow to inf→NaN) and `from_lorentz` now projects its output into the ball — making *every* `from_lorentz` consumer boundary-safe, symmetric with `to_lorentz`'s existing projection. The cap is **not** a magic literal: it is `atanh(1-BOUNDARY_EPS)`, derived from BOUNDARY_EPS and pinned by `exp_saturation_cap_matches_boundary_eps`. So `sqrt(k)||x|| ≤ 1-BOUNDARY_EPS` holds at the source for all decode primitives; no downstream rescale shim.

**Comprehensive regression tests.** Rust: `exp_origin_large_tangent_stays_strictly_interior`, `lorentz_exp_origin_large_tangent_is_finite_and_interior`, and `decode_large_gates_stay_strictly_interior_both_paths` cover both paths across curvatures {-1, -0.25, -4}, asserting finiteness, strict interiority, exact landing at max_norm, and finite distance. Python: `tests/test_poincare_atoms.py::test_large_gate_decode_stays_strictly_interior` and `::test_exp_origin_large_tangent_is_finite_and_interior` run the exact issue repro (`pa(torch.full((3,4), 1e3))`) on both Poincaré and Lorentz decoders plus the direct Rust primitives. The FD pin `tests/poincare_tangent_decode_backward_fd.rs` guards the gradient. Edge cases (origin floor via ORIGIN_EPS, both curvature signs) handled.
